src/main/java/Deadline.java

```diff
@@ -0,0 +1,50 @@
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Defines a Deadline type task.
+ *
+ * @author Kai Chao
+ * @version 1.0
+ * @since 26-08-2020
+ */
+public class Deadline extends Task {
+
+    protected LocalDate localDate;
+    protected LocalTime localTime;
+
+    Deadline(String description, String localDate, String localTime) {
+        super(description);
+        this.localDate = localDate != "" ? LocalDate.parse(localDate) : null;
+        this.localTime = localTime != "" ? LocalTime.parse(localTime) : null;
+    }
+
+    public boolean hasDate() {
+        return this.localDate != null;
```

Comment:
Should you remove the this keyword? The [java coding standard documentation](https://se-education.org/guides/conventions/java/index.html#variables) mentions to use this keyword only when a field is shadowed by a method or constructor parameter. I've noticed this issue in several other places as well.

Reply:
Missed out on that standard, thanks!